### PR TITLE
Close issue #9823 Make users aware that download was not performed because of a denied permission

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/downloads/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -45,4 +45,6 @@
     <string name="mozac_feature_downloads_third_party_app_chooser_dialog_title">Complete action using</string>
     <!-- Message that appears when trying to download with an external app fails. %1$s will be replaced with the name the external app. -->-->
     <string name="mozac_feature_downloads_unable_to_open_third_party_app">Unable to open %1$s</string>
+    <!-- Text for the info dialog when write to storage permissions have been denied but user tries to download a file. -->
+    <string name="mozac_feature_downloads_write_external_storage_permissions_needed_message">Storage access needed to download files. Go to Android settings, tap permissions, and tap allow.</string>
```
Comment (csadilek) [inline components/feature/downloads/src/main/res/values/strings.xml:49]: nit: Should we mention or hint at which permission to allow? Tap allow is quite generic...users need to know it's the files and media permission.

File: components/support/base/src/main/java/mozilla/components/support/base/dialog/DeniedPermissionDialogFragment.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import androidx.annotation.StringRes
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.app.AlertDialog
+import androidx.fragment.app.DialogFragment
+import mozilla.components.support.base.R
+
+internal const val KEY_MESSAGE = "KEY_MESSAGE"
+
+/**
+ * An dialog to display when Android permission ise denied and
```
Comment (csadilek) [inline components/support/base/src/main/java/mozilla/components/support/base/dialog/DeniedPermissionDialogFragment.kt:21]: nit: A dialog displayed when. the required Android permission...
nit: s/ise/is

File: components/support/base/src/main/java/mozilla/components/support/base/dialog/DeniedPermissionDialogFragment.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ * you want give users a way activate it on the app settings.
```
Comment (csadilek) [inline components/support/base/src/main/java/mozilla/components/support/base/dialog/DeniedPermissionDialogFragment.kt:22]: nit: users should be notified and offered a way to ....

**Review by csadilek — APPROVED** [review]
Tested locally. Works great, just some nits below.

File: components/feature/downloads/src/main/res/values/strings.xml (hunk 1 above)
Comment (Amejia481) [reply components/feature/downloads/src/main/res/values/strings.xml:49]: Ok maybe we could use "files and media permission access needed to download files. Go to Android settings, tap permissions, and tap allow."

**Comment by Amejia481** [comment]
Thanks for the review, I addressed the comments.

**Pull request merged**